java/ql/src/experimental/Security/BigDecimalDOS/BigDecimalDOS.ql

```diff
@@ -0,0 +1,38 @@
+/**
+ * @id java/BigDecimalDOS
+ * @name Java-BigDecimal-DOS-Vulnerability
+ * @description When user-controllable data is pass to the relevant Methods in BigDecimal, it may cause DOS issues when computing resources are limited. This issue is common in business scenarios such as e-commerce platforms.
+ * @kind path-problem
+ * @problem.severity error
+ */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import DataFlow::PathGraph
+
+class BigDecimalDOSConfig extends TaintTracking::Configuration {
+  BigDecimalDOSConfig() { this = "Java-BigDecimal-DOS-Vulnerability-Config" }
+
+  override predicate isSource(DataFlow::Node src) { src instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(Method method, MethodAccess call |
```

Comment:
Have you considered adding other methods of the `BigDecimal` class that could be equally problematic, e.g. [`divide`](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#divide-java.math.BigDecimal-), [`multiply`](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#multiply-java.math.BigDecimal-) or [`pow`](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#pow-int-)?

Reply:
Thanks @atorralba for helping review this part of the code. 
Yes, I will do more tests and add these methods!